File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        // instead of string each charachter 
        // store the index of previous occurance
        
        vector<int>id1(200,-1);
        vector<int>id2(200,-1);
        
        for(int i=0;i<s.size();i++)
        {
            if(id1[s[i]]!=id2[t[i]])
                return 0;
            
            id1[s[i]]=i+1;
            id2[t[i]]=i+1;
        }
        
        return 1;
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        int []id1=new int[200];
        int []id2=new int[200];
        
        for(int i=0;i<s.length();i++)
        {
            if(id1[s.charAt(i)]!=id2[t.charAt(i)])
                return false;
            
            id1[s.charAt(i)]=i+1;
            id2[t.charAt(i)]=i+1;
        }
        return true;
    }
}